src/NMEA/EngineState.hpp

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The XCSoar Project
+
+#pragma once
+
+#include "Validity.hpp"
+#include "time/Stamp.hpp"
+#include "Atmosphere/Temperature.hpp"
+
+/** Information about the engine sensor box values,
+ *  temperatures and revolutions per second.
+ *  Update rate of the engine sensor box is 1 Hz.
+*/
+struct EngineState
+{
+  /** Just one of the 8 bit is used as Flag.
+   *  Lowest bit 0b00000001 = 0x1 set to 1, indicates a valid RPM value.
+   *  All other bits are reseved for further use.
+  */
+  Validity sensbox_flags_available;
+  uint8_t sensbox_flags;
+
+  // The engine box measured revolutions per second.
+  Validity revs_per_sec_available;
+  uint16_t revs_per_sec;
+
+  /** temp1 is a generic temperature sensor. The InfoBox showing
+   *  this value has a user configurable title. The user can choose
+   *  3 different titles: {CHT, EGT, WTR}
```

Comment:
What the .... is CHT EGT WTR? (WTF)
Why does the user have to choose that? Why two temperature sensors? 
What if some piece of XCSoar code needs to have the "EGT" temperature - how will it obtain that?
Your design is bad because it requires complexity everywhere the value is used. A better design could be:
```
Temperatur cht_temperature, egt_temperature, wtr_temperature;
```
Then the driver has to figure out how to fill these fields. Not the UI code.
The UI code doesn't know about the device configuration, and should not care. All it has to do is well-defined read values from NMEAInfo.

Reply:
Look, this design lets the user of xcs choose for different meanings of temp1 and temp2, depending on the wiring and use of the temperature sensors on the engine. In my mind the user should have the flexibility to enter arbitrary text for the InfoBox titles for temp1 and temp2, but I think you would not agree, right? But let the user choose would be most generic.  It could then be for instance temp1 = BAT = Battery 1 of a electric engine, or temp1 = WTR = Water Temperature of a water cooled combustion engine and so forth...

So my suggestion would be: Let it be temp1 & temp2 in the code. Let the user of xcs choose title text for it.

Reply:
No, this is a bad design, and I have explained why. You did not address my criticism, and you didn't answer my question.

Reply:
Okay, so I create variables for cht_temperature (1 cylinder machine) , egt_temperature, wtr_temperature, bat_temperature, egt2_temperature (second cylinder egt), cht2_temperature (second cylinder), ee_temperature (electrical engine temperature), esc_temperature (electric speed controller) and so on?

Reply:
It will be obtained by a cold junction compensated MAX6675 type-K thermoelement. Two of those will measure temp1 & temp2

Reply:
Your documentation explicitly said there are only 3 choices, but now you present 8 choices This is a new story. What's up?

Reply:
> It will be obtained by a cold junction compensated MAX6675 type-K thermoelement. Two of those will measure temp1 & temp2

Is that supposed to be an answer to one of my questions? I don't understand.

Reply:
Okay, I surrender. No user option. Just cht_temperature and egt_temperature. I don't understand either...

Reply:
I think it is not really clear what your box exactly does. Is it a more generic box with two temperature readings that "could" potentially anywhere? Or is your box more tied to a specific engine model (with fixed EGT and CHT readings) and other boxes "might" use WTR, BAT, etc... later? Based on that you have different design choices for the software.

Reply:
I tried to explain... Now it is just for RPM, CHT and EGT. Specific engine.  No choices for the software